pkg/infra/httpclient/httpclientprovider/azure_middleware.go

```diff
@@ -82,20 +81,33 @@ func getDefaultAzureCredentials(cfg *setting.Cfg) azcredentials.AzureCredentials
 	}
 }
 
-func getAzureEndpointScopes(customOptions map[string]interface{}) ([]string, error) {
-	if untypedValue, ok := customOptions["azureEndpointResourceId"]; !ok {
+func getAzureEndpointResourceId(customOptions map[string]interface{}) (*url.URL, error) {
+	var value string
+	if untypedValue, ok := customOptions["_azureEndpointResourceId"]; !ok {
```

Comment:
the `_` prefix is a bit weird, I am not sure why these are needed but if they are I would just use `az` or something like that as a prefix (the underscore for me it means that it's hidden or something).

Reply:
My expectation is that `CustomOptions` should be only temporary solution until we get a proper strongly typed model (discussion #37044).

All contents of `JsonData` is being dumped into `CustomOptions` ([here](https://github.com/grafana/grafana/blob/0bf1a97262c6fd4d64dccd806147f58ba35de18b/pkg/models/datasource_cache.go#L109)), but since I do type transformation (from key-value pairs to actual struct) I need control of what values set in `CustomOptions`, so I decided to use underscores to distinguish original values which came from `JsonData` to values which I set myself (while keeping names related).

Again, this is until we have strongly typed models.